pkg/traceql/ast.go

```diff
@@ -303,6 +306,23 @@ func (s Static) impliedType() StaticType {
 	return s.Type
 }
 
+func (s Static) Equals(other Static) bool {
+	return s.Type == other.Type && s.N == other.N && s.F == other.F && s.S == other.S && s.B == other.B && s.D == other.D && s.Status == other.Status
```

Comment:
This could be `s == other` unless I am missing something.  However there is likely special logic needed here for span `status` that came up in #1806.  We have `TypeStatus` to represent `unset, ok, error` special values, but it can also be addressed as an int `status = 42`, which impacts the equality logic here.

Reply:
So If I'm understanding you correctly, it should also be valid to compare a status with an int? So we can do a simple `s == other` unless type is `TypeStatus` in which we also check if `s.Status = other.N`.